File: Week 1/problem1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void) 
{
    //Get the inpur from the user
    int n = get_int("Enter the number of blocks: ");
    //User input validation: checks if the value is positive and it's in the rage from 1 to 8 only
    while (n < 1 || n > 8)
    {
        n = get_int("Enter the number of blocks: ");
    }
    //This loop creates lines with n number of # with a space before each hash
    for (int i = 0; i < n; i++)
    {
        //This loop creates n - 1 number of hashes and it iterates as long as k's value is larger than i
        for (int k = n - 1; k > i ; k--)
        {
            printf(" ");
        }
        //This loop prints a hash in the end of each line
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        //Creates a new line at the end of each iteration of the for (int i=0; i<n; i++) loop
        printf("\n");
    }
}

File: Week 6/problem3/mario.py
# Get number of blocks from the user
# Check if the number of blocks is in range 1 to 8
# If it doesn't satisfy the condition prompt the user again
while True:
    try:
        n = int(input("Enter the number of blocks: "))
    except ValueError:
        pass
    else:
        break
while n < 1 or n > 8:
    n = int(input("Enter the number of blocks: "))

if n == 0:
    exit()

else:
    for i in range(n):
        k = n - 1
        # This loop creates n - 1 number of hashes and it iterates as long as k's value is larger than i
        while k > i:
            print(" ", end='')
            k -= 1
        j = 0
        # This loop prints a hash in the end of each line
        while j <= i:
            print("#", end='')
            i -= 1
        print("")